src/machine/machinectl.c

```diff
@@ -1668,6 +1668,8 @@ static int enable_machine(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
+        unit_file_changes_free(changes, n_changes);
+
```

Comment:
This doesn't look right... the changes array needs to be freed in case of failure too. i figure that means you have to add a "fail:" section to the end, and make sure all error paths jump to it instead of returning directly.